staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -123,7 +123,7 @@ func createHandler(r rest.NamedCreater, scope *RequestScope, admit admission.Int
 			scope.err(err, w, req)
 			return
 		}
-		if gvk.GroupVersion() != gv {
+		if gvk.GroupVersion() != gv && (scope.AcceptKind == nil || !scope.AcceptKind.AcceptGroupVersionKind(*gvk)) {
```

Comment:
Maybe do the nil check first and then decide which check to do based on that?

Reply:
done, moved to a function in scope as well